### PR TITLE
Use sewing-kit to test, format, and lint

### PR DESCRIPTION
**Pull request by kaelig**

### CODE REVIEW

**Comment by kaelig** [comment]
@BPScott I'm confused as to why `yarn dist:format` doesn't format files in the dist/ directory. Any pointers?

File: package.json [hunk 1]
```diff
@@ -50,16 +50,17 @@
       "plugin:shopify/prettier"
     ]
   },
+  "stylelint": {
+    "extends": [
+      "stylelint-config-shopify"
```
Comment (BPScott) [inline package.json]: You should extend from  "stylelint-config-shopify/prettier" if you want to enable running prettier as part of stylelint

File: package.json [hunk 2]
```diff
@@ -14,13 +14,13 @@
...
-    "dist:colors:prettier-sketchpalette": "prettier ./dist/*.sketchpalette --write --parser json",
+    "dist:format": "sewing-kit format",
+    "dist:colors:format-sketchpalette": "prettier ./dist/*.sketchpalette --write --parser json",
     "dist:colors:ase": "node ./scripts/ase-encode.js",
     "dist:colors:clr:exclude-from-ci": "ase2clr ./dist/colors.ase ./dist/colors.clr",
     "dist:colors:rename-palettes:exclude-from-ci": "renamer --find colors --replace Polaris ./dist/colors.{ase,clr,sketchpalette}",
     "lint": "npm-run-all lint:**",
-    "lint:js": "eslint . --format codeframe",
+    "lint:js": "sewing-kit lint",
     "lint:tokens": "yamllint ./tokens/*.yml",
```
Comment (BPScott) [inline package.json]: Prettier should now run over yml files. Does yamllint offer anything that prettier does not?

File: package.json [hunk 3]
```diff
@@ -72,19 +73,20 @@
...
-    "prettier": "1.15.2",
     "renamer": "^1.1.0",
     "rimraf": "^2.6.2",
     "run-sequence": "^2.2.1",
     "theo": "8.0.0",
     "tinycolor2": "^1.4.1",
     "xml": "^1.0.1",
     "yaml-lint": "^1.2.4"
+  },
+  "dependencies": {
```
Comment (BPScott) [inline package.json]: Should be a devDependency

File: package.json (hunk 2 above)
Comment (kaelig) [reply package.json]: Good point - I just removed yaml-lint!

File: dist/index.custom-properties.css [hunk 4]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (kaelig) [inline dist/index.custom-properties.css]: Not sure why .css files don't get formatted correctly. Any idea?

File: docs/index.css [hunk 5]
```diff
@@ -5,18 +5,13 @@ body {
...
-    'Segoe UI', 'Helvetica Neue', sans-serif;
-}
+  font-family: -apple-system, 'BlinkMacSystemFont', 'San Francisco', 'Roboto', 'Segoe UI', 'Helvetica Neue', sans-serif; }
 
 @media (min-width: 40em) {
   html,
   body {
-    font-size: 1.4rem;
-  }
-}
```
Comment (BPScott) [inline docs/index.css]: Where are these trailing bracket changes coming from? Can they be reverted?

File: package.json [hunk 6]
```diff
@@ -14,14 +14,12 @@
     "dist-ci": "npm-run-all dist:* dist:*:*",
     "dist:gulp": "gulp",
     "dist:docs": "gulp docs",
-    "dist:prettier": "prettier ./{dist,docs}/*.{scss,css,json,js} --write",
-    "dist:colors:prettier-sketchpalette": "prettier ./dist/*.sketchpalette --write --parser json",
+    "dist:format": "sewing-kit format",
+    "dist:colors:format-sketchpalette": "prettier ./dist/*.sketchpalette --write --parser json",
```
Comment (BPScott) [inline package.json]: I think keep the overrides section on the prettierrc, then you can remove the `--parser` here.

File: dist/index.custom-properties.css (hunk 4 above)
Comment (BPScott) [reply dist/index.custom-properties.css]: css files aren't included in sewing-kit's glob when calling stylelint https://github.com/Shopify/sewing-kit/blob/master/src/tools/stylelint/index.ts#L38

File: package.json (hunk 6 above)
Comment (kaelig) [reply package.json]: Good point!

File: docs/index.css (hunk 5 above)
Comment (kaelig) [reply docs/index.css]: Yep - that was due to the fact sewing-kit didn't format css files.

**Comment by kaelig** [comment]
@BPScott looks like this is now building, linting, and generally working correctly!

Care to have a last look?

File: package.json [hunk 7]
```diff
@@ -14,14 +14,11 @@
     "dist-ci": "npm-run-all dist:* dist:*:*",
     "dist:gulp": "gulp",
     "dist:docs": "gulp docs",
-    "dist:prettier": "prettier ./{dist,docs}/*.{scss,css,json,js} --write",
-    "dist:colors:prettier-sketchpalette": "prettier ./dist/*.sketchpalette --write --parser json",
+    "dist:format": "prettier ./{dist,docs}/*.{js,json,css,scss,sketchpalette} --write",
```
Comment (kaelig) [inline package.json]: feels weird to call prettier when it's not listed in the devDependencies. Thoughts?

Comment (BPScott) [reply package.json]: I'm fine with using it as a transitive dependency. It is preferable to specifying prettier in your package.json and having to keep it in lock-step with the version sewing-kit provides to make sure you don't install two differing versions. I think this is a better solution than avoiding using sewing-kit at all

**Comment by kaelig** [comment]
> I've already said I dislike running prettier over dist folders

Some projects, like polaris-react, copy tokens in a folder that gets linted. This means that offending files throw errors if formatted incorrectly. Granted, these projects could tweak the linting settings to ignore the copied folder altogether.

> if you want it pretty printed you should fix whatever tool generates it

The code required to output nicely printed files would dilute the intent of the format itself, and require maintenance every time our Prettier config changes.

Some of the formatting is inherited from the `theo` package (e.g. [common.js format](https://github.com/salesforce-ux/theo/blob/master/lib/formats/common.js.js)), meaning we'd also have to copy that code and tweak it to match our preferences. I'd rather use what this tool provides out of the box as much as possible.

The alternative would be to invoke Prettier via a Gulp task (at generation time) instead of an npm script?

**Comment by BPScott** [comment]
> The alternative would be to invoke Prettier via a Gulp task (at generation time) instead of an npm script?

I think my root concern is "generating the dist folder should be a one command process". Looking deeper it seems like we already do that with `yarn run dist`, I just wasn't expecting the separate steps to be defined as multiple package.json commands.

File: package.json (hunk 7 above)
Comment (BPScott) [reply package.json]: Extra thought: How do we guarantee this runs as the final step when running `yarn run dist`? I think previously it was dumb-luck that the` dist:prettier` was last alphabetically. Would changing the name have an effect?

**Comment by kaelig** [comment]
👍 

I admit that the mix between all these tools (npm, prettier, sewing-kit, gulp) can be confusing 😓

**Pull request merged**